account_invoice_triple_discount/migrations/16.0.2.0.0/pre-migrate.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from openupgradelib import openupgrade
+
+
+def migrate_discount_to_discount1(env):
+    openupgrade.add_fields(
+        env,
+        [
+            (
+                "discount1",
+                "account.move.line",
+                "account_move_line",
+                "float",
+                "numeric",
+                "account_invoice_triple_discount",
+                0.0,
+            )
+        ],
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move_line
+        SET discount1 = discount;
+        """,
+    )
+    # if discounts are : 10% - 20% - 30% main discount is : 49.6 %
+    # if discounts are : 05% - 09% - 13% main discount is : 24.7885 %
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move_line
```

Comment:
@legalsylvain Your script doesn't take into account the discount_type to compute the discount value....

Reply:
Hi @lmignon. could you elaborate ? I don't understand what is discount_type.
**In V16 :** 
<code>
$ rgrep "discount_type" ./ --include=*.py
./addons/l10n_it_edi/models/account_edi_format.py: discount_type = discount_element.xpath('.//Tipo')
./addons/l10n_it_edi/models/account_edi_format.py: if discount_type and discount_type[0].text == 'MG':
./addons/l10n_it_edi/models/account_edi_format.py: discount_type = discount_element.xpath('.//Tipo')
./addons/l10n_it_edi/models/account_edi_format.py: if discount_type and discount_type[0].text == 'MG':
./addons/l10n_it_edi/models/account_invoice.py: def discount_type(discount):
./addons/l10n_it_edi/models/account_invoice.py: 'discount_type': discount_type,
</code>